lib/ngzip/builder.rb

```diff
@@ -81,7 +84,8 @@ def file_list(files)
       Array(files).map do |e|
         if File.directory?(e)
           # `expand_path` removes any trailing slash from the path string
-          sanitized_path = File.expand_path(e)
+          # `String#escape_glob` handles bracket literals otherwise interpreted as glob control characters
+          sanitized_path = File.expand_path(e.escape_glob)
```

Comment:
Previously since the `[]` characters were not escaped the following `Dir.glob` would not correctly inspect the directory for its contents.